Imagekit.UnitTests/ImageKitRequestModelValidation.cs

```diff
@@ -0,0 +1,711 @@
+
+            var client = mockHttp.ToHttpClient();
+            RestClient rs = new RestClient(GOOD_PRIVATEKEY, GOOD_URLENDPOINT, client);
+            var response = rs.CopyFile(model);
+            mockHttp.VerifyNoOutstandingExpectation();
+        }
+
+        [Fact]
+        public void MoveFile_ModelValidation()
+        {
+            string reqJosn = "{\"sourceFilePath\":\"Tst3\",\"destinationPath\":\"Tst3\"}";
+            MoveFileRequest model = new MoveFileRequest
+            {
+                sourceFilePath = "Tst3",
+                destinationPath = "Tst3"
+            };
+
+            string url = string.Empty;
+            url = string.Format("https://api.imagekit.io/v1/files/move");
+            var mockHttp = new MockHttpMessageHandler();
+            var request = mockHttp.Expect(url)
+                .With(a => a.Method.Equals(HttpMethod.Post))
+                .WithHeaders("Authorization: Basic cHJpdmF0ZV9rZXk6")
+                .WithContent(reqJosn)
+                .Respond("application/json", "{'name' : 'ImageKit Response'}");
+
+            var client = mockHttp.ToHttpClient();
+            RestClient rs = new RestClient(GOOD_PRIVATEKEY, GOOD_URLENDPOINT, client);
+            var response = rs.MoveFile(model);
+            mockHttp.VerifyNoOutstandingExpectation();
+        }
+
+        [Fact]
+        public void RenameFile_ModelValidation()
+        {
+            string reqJosn = "{\"filePath\":\"Tst3\",\"newFileName\":\"Tst4\",\"purgeCache\":false}";
+            RenameFileRequest model = new RenameFileRequest
+            {
+                filePath = "Tst3",
+                newFileName = "Tst4",
+                purgeCache = false
+            };
+
+            string url = string.Empty;
+            url = string.Format("https://api.imagekit.io/v1/files/rename");
+            var mockHttp = new MockHttpMessageHandler();
+            var request = mockHttp.Expect(url)
+                .With(a => a.Method.Equals(HttpMethod.Put))
+                .WithHeaders("Authorization: Basic cHJpdmF0ZV9rZXk6")
+                .WithContent(reqJosn)
+                .Respond("application/json", "{'name' : 'ImageKit Response'}");
+
+            var client = mockHttp.ToHttpClient();
+            RestClient rs = new RestClient(GOOD_PRIVATEKEY, GOOD_URLENDPOINT, client);
+            var response = rs.RenameFile(model);
+            mockHttp.VerifyNoOutstandingExpectation();
+        }
+
+
+        [Fact]
+        public void CopyFolder_ModelValidation()
+        {
+            string reqJosn = "{\"sourceFolderPath\":\"Tst3\",\"destinationPath\":\"Tst3\",\"includeFileVersions\":false}";
+            CopyFolderRequest model = new CopyFolderRequest
+            {
+                sourceFolderPath = "Tst3",
+                destinationPath = "Tst3"
+
+            };
+            string url = string.Format("https://api.imagekit.io/v1/bulkJobs/moveFolder");
```

Comment:
`rs.CopyFolder(model)` is calling moveFolder endpoint. This will create huge issues for customers.